### ISSUE TITLE
Serval HiDPI issues on Linux and Windows

### ISSUE THREAD

**Issue by jonasschnelli** [bitcoin/bitcoin#6472]
Somehow I expected that Qt upstream does handle the HiDPI differences. It looks like that handling HiDPI correctly would require some streamlining in our code.

Splashscreen

HiDPI Window8 NOK
<img width="1182" alt="bildschirmfoto 2015-07-24 um 11 38 39" src="https://cloud.githubusercontent.com/assets/178464/8872301/00dfe0ca-31fc-11e5-94b0-13dbb4f8e0cd.png">

Linux NOK:
![bildschirmfoto-2015-07-24-um-12 16 11](https://cloud.githubusercontent.com/assets/178464/8872504/d9cf262e-31fd-11e5-8658-b3cf606b8888.jpg)

OSX OK:
<img width="564" alt="bildschirmfoto 2015-07-24 um 12 04 31" src="https://cloud.githubusercontent.com/assets/178464/8872321/30b56432-31fc-11e5-8f4d-b357e98a36ec.png">

Icon big/small mix:
Linux NOK:
<img width="526" alt="bildschirmfoto 2015-07-24 um 12 14 09" src="https://cloud.githubusercontent.com/assets/178464/8872469/8d2c6340-31fd-11e5-9f6e-1b5ca005c8f8.png">

OSX OK:
<img width="1035" alt="bildschirmfoto 2015-07-24 um 12 15 29" src="https://cloud.githubusercontent.com/assets/178464/8872486/b4e38760-31fd-11e5-8031-86021a45da0a.png">

Linux Menu Icons wrong size:
![bildschirmfoto-2015-07-24-um-12 17 12](https://cloud.githubusercontent.com/assets/178464/8872517/fe928564-31fd-11e5-9937-812a42787035.jpg)

**Comment by jonasschnelli** [comment]
Maybe we should disable HiDPI for Win/Linux until it's streamlines upstream?
https://github.com/bitcoin/bitcoin/blob/master/src/qt/bitcoin.cpp#L514

**Comment by theuni** [comment]
If [this](http://lists.qt-project.org/pipermail/development/2015-June/021631.html) is to be believed, then disabling for !osx makes sense at least until we can test with 5.6.

**Comment by theuni** [comment]
@jonasschnelli I couldn't help myself, so I tried out building the qt 5.6 wip branch with dpi changes: https://github.com/qtproject/qtbase/tree/wip/highdpi

I'm not sure if there are changes necessary in downstream code, or if things are expected to "just work".

I can't actually test it, but here are some binaries if you're interested.
Linux: https://bitcoincore.org/cfields/bitcoin-qt-linux-qt56-hidpi
Win: https://bitcoincore.org/cfields/bitcoin-qt-win-qt56-hidpi.exe

It's my current qt "5.5 bump" branch with this hack on top:
```diff
diff --git a/depends/packages/qt.mk b/depends/packages/qt.mk
index 0fd3f32..440b65a 100644
--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -1,9 +1,10 @@
 PACKAGE=qt
 $(package)_version=5.5.0
-$(package)_download_path=http://download.qt.io/official_releases/qt/5.5/$($(package)_version)/submodules
+$(package)_download_path=https://github.com/qtproject/qtbase/archive
+$(package)_extra_download_path=http://download.qt.io/official_releases/qt/5.5/$($(package)_version)/submodules
 $(package)_suffix=opensource-src-$($(package)_version).tar.gz
-$(package)_file_name=qtbase-$($(package)_suffix)
-$(package)_sha256_hash=7e82b1318f88e56a2a9376e069aa608d4fd96b48cb0e1b880ae658b0a1af0561
+$(package)_file_name=5bfb6ce4bf5b090dc88463a2e7a4aae3ef10fe09.tar.gz
+$(package)_sha256_hash=6ca060d1c133f142ce29383e1dec7118198d97c86db7f73e0890d9ac80a5c482
 $(package)_dependencies=openssl
 $(package)_linux_dependencies=freetype fontconfig dbus libxcb libX11 xproto libXext
 $(package)_build_subdir=qtbase
@@ -102,8 +103,8 @@ endef
 
 define $(package)_fetch_cmds
 $(call fetch_file,$(package),$($(package)_download_path),$($(package)_download_file),$($(package)_file_name),$($(package)_sha256_hash)) && \
-$(call fetch_file,$(package),$($(package)_download_path),$($(package)_qttranslations_file_name),$($(package)_qttranslations_file_name),$($(package)_qttranslations_sha256_hash)) &
-$(call fetch_file,$(package),$($(package)_download_path),$($(package)_qttools_file_name),$($(package)_qttools_file_name),$($(package)_qttools_sha256_hash))
+$(call fetch_file,$(package),$($(package)_extra_download_path),$($(package)_qttranslations_file_name),$($(package)_qttranslations_file_name),$($(package)_qttranslations_sha256_ha
+$(call fetch_file,$(package),$($(package)_extra_download_path),$($(package)_qttools_file_name),$($(package)_qttools_file_name),$($(package)_qttools_sha256_hash))
 endef
 
 define $(package)_extract_cmds
@@ -121,6 +122,7 @@ define $(package)_extract_cmds
 endef
 
 define $(package)_preprocess_cmds
+  mkdir -p qtbase/.git && \
   sed -i.old "s|updateqm.commands = \$$$$\$$$$LRELEASE|updateqm.commands = $($(package)_extract_dir)/qttools/bin/lrelease|" qttranslations/translations/translations.pro && \
   sed -i.old "s/src_plugins.depends = src_sql src_xml src_network/src_plugins.depends = src_xml src_network/" qtbase/src/src.pro && \
   sed -i.old "s/PIDLIST_ABSOLUTE/ITEMIDLIST */" qtbase/src/plugins/platforms/windows/qwindowscontext.h &&\
```

**Comment by jonasschnelli** [comment]
Hmm.. tested shortly both versions (linux/win). No difference visible.

<img width="474" alt="bildschirmfoto 2015-07-25 um 02 02 21" src="https://cloud.githubusercontent.com/assets/178464/8886884/f690b31c-3271-11e5-8b3d-54ae347283df.png">
<img width="1182" alt="bildschirmfoto 2015-07-25 um 02 06 51" src="https://cloud.githubusercontent.com/assets/178464/8886885/f6d0667e-3271-11e5-978d-7706745cf5f3.png">
<img width="918" alt="bildschirmfoto 2015-07-25 um 02 01 58" src="https://cloud.githubusercontent.com/assets/178464/8886888/0624f77a-3272-11e5-9b90-5c002f582742.png">

**Comment by jonasschnelli** [comment]
I found out by running `bitcoin-qt` with the environment variable `QT_DEVICE_PIXEL_RATIO=2` looks much better on HiDPI linux, thought, not perfect and not the same as on OSX.

**Comment by sipa** [comment]
I don't understand why this is causing problems. In what way is HiDPI not just doubling the dpi of fonts and scaling everything proportionally?

**Comment by jonasschnelli** [comment]
@sipa: If you have a 2880x1800 screen, you barely can't see anything if you don't *scale the UI*. Therefore apple announced the *2 approach. Just double everything, image/icon size, window bars, font, etc. This gives a extremely comfortable result.

Now the problem is, that on Linux, Qt auto-scale the Font and metrics, but leaving out the Icons in Buttons as well as window sizes.
IMO it's just not finished or fixed in all places (QT upstream).

**Comment by sipa** [comment]
I would have expected that higher DPI would scale all UI elements too proportionally. That seems like a much more flexible solution...

**Comment by jonasschnelli** [comment]
Proportional, non-fixed-step scaling makes sense.. modern linux desktops allows that. It would work if Qt would scale the icons/window sizes according to the given scale factor.

**Comment by laanwj** [comment]
> I don't understand why this is causing problems. In what way is HiDPI not just a higher DPI setting?

It's a typical theory versus practice thing. If UI systems had been designed from the start to be scalable instead of adapted to a selection of resolutions it'd all have been a lot easier. Now it causes issues on the whole spectrum from the OS itself to GUI libraries to applications themselves, as well as malfunctions due to communication issues between those levels.

**Comment by jonasschnelli** [comment]
Upstream issue, should be better with Qt5.6 (once it's released, planed for Q1/2016):
http://blog.qt.io/blog/2015/12/18/qt-5-6-beta-released/

**Comment by fanquake** [comment]
@jonasschnelli What's the status here now that we've switched to Qt 5.6.1 ?

**Comment by MarcoFalke** [comment]
Duplicate of #7341?

**Comment by laanwj** [comment]
Yes, I think it's solved with the switch to Qt 5.6.1. @cosmicvibe reported that in #7341 after installing Qt 5.6.1 the text was readable.

**Comment by laanwj** [comment]
Closing due to Qt 5.6.1 switch which addresses this (#8210).
